classes/output/exemption_table.php

```diff
@@ -0,0 +1,106 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace factor_exemption\output;
+
+class exemption_table extends \table_sql {
```

Comment:
Probably outside the scope of this, but I would recommend using the new moodle report builder for things like this. It has built in things for stuff like actions, filter, etc... and you can easily link the user source and get things like user profile links without writing much code.

DM me if you want an example (internal) since it is a tad confusing the first time you look at it

Reply:
(just checked and report builder in Moodle is 4.0+ so probably wouldn't work here btw since this is intended for 3.9+)

Reply:
Yeah I want to support older moodles, but its a good consideration, I hadnt considered report builder for this task. Interesting stuff